sdk/aqueduct/artifacts/utils.py

```diff
@@ -49,6 +50,19 @@ def preview_artifacts(
         make_copy=True,
     )
 
+    if globals.__GLOBAL_CONFIG__.engine is not None:
+        engine = globals.__GLOBAL_CONFIG__.engine
+        if engine is None:
+            engine_config = EngineConfig()
+        else:
+            connected_integrations = globals.__GLOBAL_API_CLIENT__.list_integrations()
+            if engine not in connected_integrations.keys():
+                raise InvalidIntegrationException(
+                    "Not connected to compute integration %s!" % engine
+                )
+            engine_config = generate_engine_config(connected_integrations[engine])
+        subgraph.engine_config = engine_config
```

Comment:
After rebasing, use `dag.set_engine_config()` here instead. That'll catch any invalid resource configuration errors too.

That PR also makes `engine_config` optional, but it has to be set before uploading to the server. You can see how this pattern works in `aqueduct_client.py`, but essentially we should be calling `set_engine_config()` regardless of whether the engine was set globally, and calling `assert subgraph.engine_config is not None` right before the api_client.preview() call as a safety check.